ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/BeaconBlockInvariants.java

```diff
@@ -36,16 +44,22 @@ public static UInt64 extractBeaconBlockSlot(final Bytes bytes) {
   }
 
   /**
-   * Extract the slot value from any BeaconBlock.
+   * Extract the slot value from any {@link SignedBeaconBlock} or {@link SignedBlockContents}.
    *
    * <p>The slot is the first field but is inside the variable length beacon block so a 4 byte
    * offset to the start of the beacon block data is recorded. Use that prefix to get the beacon
    * block data and then find the slot as for an unsigned block
    *
-   * @param bytes the signed beacon block slot to extract a slot from
+   * @param bytes the SSZ bytes to extract a slot from
    */
   public static UInt64 extractSignedBeaconBlockSlot(final Bytes bytes) {
-    final int blockDataOffset = SszType.sszBytesToLength(bytes.slice(0, 4));
+    int blockDataOffset = SszType.sszBytesToLength(bytes.slice(0, SSZ_OFFSET_SIZE));
+    if (blockDataOffset == BLOCK_DATA_OFFSET_SUGGESTS_TYPE_IS_SIGNED_BEACON_BLOCK) {
+      return extractBeaconBlockSlot(bytes.slice(blockDataOffset));
+    }
+    // extract blockDataOffset for SignedBlockContents
+    blockDataOffset =
+        blockDataOffset + SszType.sszBytesToLength(bytes.slice(blockDataOffset, SSZ_OFFSET_SIZE));
```

Comment:
I'm not sure we can do this any more given that the position isn't fixed? we could read any random ssz bytes and get a number, but the number may not be the slot...
We need to probably look at `Eth-Consensus-Version` header to know how to proceed now...

Reply:
Technically we can as long as `SignedBlockContents` container first field is the `SignedBeaconBlock`. I added a property tests to capture future milestone changes when it is not potentially and it will fail.